invenio_app_ils/records/views.py

```diff
@@ -18,8 +18,55 @@
 
 from invenio_app_ils.circulation.views import need_permissions
 from invenio_app_ils.errors import DocumentRequestError
-from invenio_app_ils.pidstore.pids import DOCUMENT_REQUEST_PID_TYPE
+from invenio_app_ils.pidstore.pids import DOCUMENT_PID_TYPE, \
+    DOCUMENT_REQUEST_PID_TYPE
 from invenio_app_ils.proxies import current_app_ils_extension
+from invenio_app_ils.signals import record_viewed
+
+
+def create_record_stats_blueprint(app):
+    """Add record stats endpoint to the blueprint."""
+    blueprint = Blueprint(
+        "invenio_app_ils_record_stats", __name__, url_prefix=""
+    )
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+    options = endpoints.get(DOCUMENT_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    rec_serializers = options.get("record_serializers", {})
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+    blueprint.add_url_rule(
+        "{0}/record-stats".format(options['item_route']),
+        view_func=RecordStatsResource.as_view(
+            RecordStatsResource.view_name,
+            serializers=serializers,
+            default_media_type=default_media_type,
+        ),
+        methods=["POST"],
+    )
+    return blueprint
+
+
+class RecordStatsResource(ContentNegotiatedMethodView):
+    """Endpoint to trigger record stats signal."""
+
+    view_name = "record_stats"
+
+    @pass_record
+    def post(self, pid, record, **kwargs):
+        """Send a signal to count record view for the record stats."""
+        data = request.get_json()
+        if data.get('stat') == 'record-view':
+            record_viewed.send(
+                current_app._get_current_object(),
+                pid=pid,
+                record=record,
+            )
+            return self.make_response(pid, record, 202)
+        return abort(400, "The proprety stat: `{}` is not supported".format(
```

Comment:
Typo `proprety`. I would say, just return 400 without much details, what do you think?
I was thinking to the 400 just because if it happens we will have a Sentry exception and we can check...